### PR TITLE
feat: provide formatters shortcuts

### LINKED ISSUE #764: feat request: provide formatters shortcuts
```ts
const p = await $`cmd arg`

const json = await $`cmd arg`.json()
const json2 = p.json()
const str = await $`cmd arg`.text()

//
.text()
.json()
.buffer()
.stream()
```

### PR DESCRIPTION
**Pull request by Anonymous961**
Fixes #764
 
1. added function for `json()`, `'text()`, `bufffer()`.
2. added tests for `json()`, `'text()`, `bufffer()`.

Not sure about the stream function. Would like to have more context. 

- [X] Tests pass
- [ ] Appropriate changes to README are included in PR

### CODE REVIEW

File: src/core.ts [hunk 1]
```diff
@@ -651,6 +661,11 @@ export function log(entry: LogEntry) {
     case 'stdout':
     case 'stderr':
       if (!entry.verbose) return
+      const eol = Buffer.from('\n', 'utf-8')
```
Comment (antongolub) [inline src/core.ts:664]: let's wait for #810

File: src/core.ts [hunk 2]
```diff
@@ -570,11 +581,10 @@ export class ProcessOutput extends Error {
...
   signal: ${inspect(this.signal)},
-  exitCode: ${(this.exitCode === 0 ? chalk.green : chalk.red)(this.exitCode)}${
-    exitCodeInfo(this.exitCode)
-      ? chalk.grey(' (' + exitCodeInfo(this.exitCode) + ')')
-      : ''
-  }
+  exitCode: ${(this.exitCode === 0 ? chalk.green : chalk.red)(this.exitCode)}${exitCodeInfo(this.exitCode)
+        ? chalk.grey(' (' + exitCodeInfo(this.exitCode) + ')')
+        : ''
+      }
```
Comment (chankruze) [inline src/core.ts:587]: Any local formatting issue ?

File: src/core.ts [hunk 3]
```diff
@@ -602,45 +612,45 @@ export async function kill(pid: number, signal?: string) {
...
+    kind: 'retry'
+    error: string
+  }
   | {
-      kind: 'custom'
-      data: any
-    }
+    kind: 'custom'
+    data: any
+  }
```
Comment (chankruze) [inline src/core.ts:653]: I think your prettier settings or something messed here.

File: src/core.ts [hunk 4]
```diff
@@ -602,45 +612,45 @@ export async function kill(pid: number, signal?: string) {
   for (const p of children) {
     try {
       process.kill(+p.pid, signal)
-    } catch (e) {}
+    } catch (e) { }
```
Comment (chankruze) [inline src/core.ts:615]: Same here and L622

**Review by chankruze — CHANGES_REQUESTED** [review]

File: src/core.ts (hunk 1 above)
Comment (Anonymous961) [reply src/core.ts]: ok I've removed this part

File: src/core.ts (hunk 3 above)
Comment (antongolub) [reply src/core.ts:653]: Let't keep the formatting: `yarn fmt`

**Review by antongolub — CHANGES_REQUESTED** [review]

File: src/core.ts (hunk 3 above)
Comment (Anonymous961) [reply src/core.ts]: Thanks @antongolub. I have now fixed the formatting.

**Review by antongolub — CHANGES_REQUESTED** [review]

File: test/core.test.js [hunk 5]
```diff
@@ -226,6 +226,21 @@ describe('core', () => {
     assert.equal((await p).toString(), 'foo\n')
   })
 
+  test('ProcessPromise: implements json()', async () => {
```
Comment (antongolub) [inline test/core.test.js:229]: Not `ProcessPromise` here but `ProcessOutput`.

Comment (Anonymous961) [reply test/core.test.js:229]: same for `text()` and `buffer()`? I just followed `toString()` test.

Comment (chankruze) [reply test/core.test.js:229]: Same for `valueOf()` also.

**Review by chankruze — CHANGES_REQUESTED** [review]

File: test/core.test.js (hunk 5 above)
Comment (Anonymous961) [reply test/core.test.js:229]: @antongolub should I also change `ProcessOutput` for `toString()` test?

**Review by antongolub — APPROVED** [review]
lgtm. we will land the rest part in another PR

**Pull request merged**